.github/workflows/_bazel.yml

```diff
@@ -62,12 +70,20 @@ jobs:
           sudo apt-get install -y --no-install-recommends clang-16 lld-16 libunwind-16 libc++abi1-16 libc++1-16 libc++-16-dev libclang-rt-16-dev llvm-16
           sed -i -e "s%llvm-symbolizer%/usr/lib/llvm-16/bin/llvm-symbolizer%" .bazelrc
       - name: Setup Windows
-        if: inputs.os_name == 'windows'
+        if: runner.os == 'Windows'
         # Set a custom output root directory to avoid long file name issues.
         run: |
           git config --global core.symlinks true
           git config --show-scope --show-origin core.symlinks
           [System.IO.File]::WriteAllLines((Join-Path -Path $env:USERPROFILE -ChildPath '.bazelrc'), 'startup --output_user_root=C:/tmp')
+      - name: Setup macOS
+        if: runner.os == 'macOS'
+        run: |
+          sudo xcode-select -s "/Applications/Xcode_15.1.app"
+          # Install lld and link it to /usr/local/bin. We overwrite any existing link, which may
+          # exist from an older pre-installed LLVM version on the runner image.
+          brew install lld
```

Comment:
Can we keep this step release-only? Installing llvm + lld takes some time, which will make builds take longer, especially for builds when the bazel cache is warm. lld provides few advantages to the test build except for slightly faster link speeds which won't make up for this (the macOS system linker has become faster in recent releases).

Reply:
> Can we keep this step release-only? 

hard to do without introducing extra switches. Also arguably we should use the same version of xcode and linker both in release and in test configuration?

Reply:
We are using the same Xcode version now that #3078 has been merged. Apple ld and lld on macOS are high-quality linkers that have very good compatibility/interop, so I think we should only spend the 80s - 90s it takes to fetch lld/LLVM for release builds, where we care more about the binary size.
You can just do `if: ${{ (if: runner.os == 'macOS') && inputs.upload_binary }}'` right? Since the Xcode flag is no longer needed the setup macOS step only contains the lld step. Then we could rename upload_binary to is_release, making it more descriptive.